src/server/pfs/server/master.go

```diff
@@ -96,7 +123,21 @@ func (d *driver) finishCommits(ctx context.Context) error {
 		repos[key] = cancel
 		go func() {
 			backoff.RetryUntilCancel(ctx, func() error { //nolint:errcheck
-				return d.finishRepoCommits(ctx, compactor, key)
+				lockCtx, err := ring.Lock(ctx, lockPrefix)
```

Comment:
We should probably have a subcontext that gets canceled in a defer to ensure resources are cleaned up during a retry.

Reply:
For this one I defined a context in the go func() but not inside the backoff, I was seeing some unnecessary cancelling when the context was defined in the backoff, but let me know if it should be changed.

Reply:
The context should be in the callback. We want to ensure the internal resources are cleaned up if finish repo commits and unlock error.

Reply:
Gotcha, will move this back and re-test